app/services/api/v1/ecf_participants/index.rb

```diff
@@ -4,58 +4,53 @@ module Api
   module V1
     module ECFParticipants
       class Index
-        attr_reader :cpd_lead_provider, :params
-
-        def initialize(cpd_lead_provider:, params:)
-          @cpd_lead_provider = cpd_lead_provider
-          @params = params
+        def initialize(lead_provider, params)
+          self.lead_provider = lead_provider
+          self.params        = params
         end
 
         def induction_records
+          join = InductionRecord
+                   .select("induction_records.id, ROW_NUMBER() OVER (PARTITION BY induction_records.participant_profile_id ORDER BY induction_records.created_at DESC) AS created_at_precedence")
+                   .joins(:participant_profile, :schedule, { induction_programme: :partnership })
+                   .where(
+                     schedule: { cohort_id: with_cohorts.map(&:id) },
+                     induction_programme: {
+                       partnerships: {
+                         lead_provider_id: lead_provider.id,
+                         challenged_at: nil,
+                         challenge_reason: nil,
+                       },
+                     },
```

Comment:
This feels complex, especially as it's inline and combining raw SQL with ActiveRecord conditions. I understand it's used here to solve a specific problem so happy to see it go in for the time being, but I think it's something to revisit.